### PR TITLE
logodds

### PR DESCRIPTION
**Pull request by mattansb**

### CODE REVIEW

**Comment by mattansb** [comment]
@DominiqueMakowski 

Should this also need an `r_to_logodds` / `d_to_logodds` / `probs_to_logodds` / `logodds_to_probs`?

Also, how do you feel about ditching all the prefixed `convert_`  aliases? 🤔

**Comment by strengejacke** [comment]
Every good effect size package has [to logodds conversion](https://github.com/strengejacke/esc) ;-)

**Comment by mattansb** [comment]
😅
Daniel, I really like your conversion functions structure (one function per outcome, instead of 1 function per conversions type). Might steal this at some point in the future.....

**Comment by DominiqueMakowski** [comment]
> Daniel, I really like your conversion functions structure (one function per outcome, instead of 1 function per conversions type). Might steal this at some point in the future.....

I think both are okay, but conversion type is super user friendly (the main problem being that the documentation is super cluttered)

**Comment by DominiqueMakowski** [comment]
@mattansb @strengejacke I added this small vignette/post as I was investigating the logodds_to_d conversion which... doesn't seem to work. Could you check my reasoning and see if my conclusions are correct or did I miss something? If so, should we think about creating an adjusted metric (kind of what I outlined in the end)?

**Comment by mattansb** [comment]
In your example you are trying to recover Cohen's d for `y ~ x` from a logistic `x ~ y`, but isn't the conversion of `log(odds) * sqrt(3) / pi` for the Cohen's d when y is dichotomize? (as in [*Effect-size indices for dichotomized **outcomes** in meta-analysis*](http://doi.org/10.1037/1082-989X.8.4.448).) 
As in recovering Cohen's d for `y ~ x` from a logistic `(y > c) ~ x`?

``` r
library(effectsize)
library(magrittr)

# I flipped the names
data <- bayestestR::simulate_difference(n=100, d=1, names=c("x", "y"))

# dichotomize the outcome X
data$y_dich <- data$y > 0

fit <- glm(y_dich ~ x, data, family = binomial())
```

these should give similar results

``` r
fit %>% 
  parameters::model_parameters() %>% 
  .[-1,] %$% 
  logodds_to_d(c(Coefficient, CI_low, CI_high)) %>% 
  setNames(c("d", "CI_low", "CI_high"))
#>         d    CI_low   CI_high 
#> 0.9342796 0.4742968 1.4213056

cohens_d(y ~ x, data = data)
#> Cohen's d |         95% CI
#> --------------------------
#>     -1.06 | [-1.48, -0.64]
```

<sup>Created on 2020-05-26 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by DominiqueMakowski** [comment]
🤯 the mindblown I was looking for, let me look at it again

**Comment by DominiqueMakowski** [comment]
wait no but here `y` is already a dichotomized outcome (simulate_difference) generates a dichotomous y and a continuous x

**Comment by mattansb** [comment]
That's because you labeled the DV as x and the group as Y (I labelled it the other way around). 
I'll try again without x/y terms (:

The conversion takes a logistic regression model where the outcome is dichotomized (a continuous `outcome` is dichotomized with some criterion `c` via `outcome_dich = outcome > c`) - the model is `outcome_dich ~ IV`, where IV is binary.
What we are trying to recover is the Cohen's d for the undichotomized outcome - the model of `outcome ~ IV`, where IV is the same binary IV as above.

What you did was try to recover Cohen's d of `outcome ~ IV` (outcome is continuous, IV is binary) from a logistic model of `IV ~ outcome`.

**Comment by DominiqueMakowski** [comment]
So you're saying in essence:

 that there if you transform to a *d* the logodds from a logistic reg (y ~ x), it's *not* supposed to be equivalent (not even in theory) to the *d* of the difference of x per groups of y (but to the difference of this *x* dichotomized, by groups of *y*?). 

If that's the case, then did I even something new? Because my algorithm for log-odds do Cohen's *d* of the reversed relationship seems to be pretty well working 😅

**Comment by mattansb** [comment]
> So you're saying in essence that if you transform to a d the logodds from a logistic reg (y ~ x), it's not supposed to be equivalent (not even in theory) to the d of the difference of x per groups of y (but to the difference of this x dichotomized, by groups of y?).

You're kinda working backwards - in the logistic model, **both y and x are binary** for this to work: x is group, y is a dichotomized continuous variable. If you take the log(odds) coefficiant of x (rep. the change in logg odds between groups) and convert it to d, this d is a "recovered" difference between groups x in the means of the continuous y.
- Y is always the DV - once continuous and once dichotomized.
- X is always binary, and is always IV.

> If that's the case, then did I even something new? Because my algorithm for log-odds do Cohen's d of the reversed relationship seems to be pretty well working 😅

Yeah, seems interesting... However, I would test how it preforms under:
- unequal variances of the continuous var between the groups
- unequal group sizes
- non normal continuous var 

I suspect these will have a great effect on the conversion you found.

(also, it will inevitably fail for large effect sizes, as you will get complete separation in the logistic model.)

File: WIP/eta_squared_posterior.R [hunk 1]
```diff
@@ -31,13 +28,14 @@ eta_squared_posterior <- function(model, partial = FALSE, nsamps = 1000, verbose
...
     # fit a simple linear model
-    temp_fit <- lm(f, temp_dat)
+    temp_fit <- stats::lm(f, temp_dat)
 
     # compute effect size
     ANOVA <- car::Anova(temp_fit, type = 3)
     es <- eta_squared(ANOVA, ci = NA, partial = FALSE)
 
-    data.frame(t(setNames(es$Eta_Sq, es$Parameter)), check.names = F)
+    es <- setNames(es$Eta_Sq, es$Parameter)
```
Comment (strengejacke) [inline WIP/eta_squared_posterior.R:37]: `stats::setNames()`

**Pull request merged**